service/src/main/kotlin/fi/espoo/evaka/nekku/NekkuQueries.kt

```diff
@@ -569,3 +570,75 @@ fun Database.Read.getNekkuSpecialDietChoices(childId: ChildId): List<NekkuSpecia
             )
         }
         .toList<NekkuSpecialDietChoices>()
+
+fun Database.Read.getNekkuOrderReport(
+    daycareId: DaycareId,
+    groupId: GroupId,
+    date: LocalDate,
+): List<NekkuOrdersReport> =
+    createQuery {
+            sql(
+                "SELECT delivery_date, daycare_id, customer_group_id, meal_sku, total_quantity, meal_time, meal_type, meals_by_special_diet FROM nekku_orders_report WHERE daycare_id = ${bind(daycareId)} AND customer_group_id = (${bind(groupId)}) AND delivery_date = ${bind(date)} "
+            )
+        }
+        .toList<NekkuOrdersReport>()
+
+fun Database.Transaction.setNekkuReportOrderReport(
+    nekkuOrders: NekkuClient.NekkuOrders,
+    groupId: GroupId,
+    nekkuProducts: List<NekkuProduct>,
+) {
+
+    val daycareId = getDaycareIdByGroup(groupId)
+
+    for (item in nekkuOrders.orders.first().items) {
+
+        val product = nekkuProducts.find { it.sku == item.sku } ?: error("Product.sku")
+
+        val mealsBySpecialDiet = item.productOptions?.map { it.value }
+
+        val report =
```

Comment:
Näistä voisi tehdä listan ja sitten tallettaa sen yhtenä batch-querynä.

Toimisiko esim.

val reportRows = nekkuOrders.orders.first().items.map {
    val product = nekkuProducts.find { it.sku == item.sku } ?: error("Product.sku")
    NekkuOrdersReport(
        LocalDate.parse(nekkuOrders.orders.first().deliveryDate),
        daycareId,
        groupId,
        item.sku,
        item.quantity,
        product.mealTime,
        product.mealType,
        item.productOptions?.map { it.value }
    )
}